fw/http.c

```diff
@@ -3029,6 +3029,109 @@ tfw_h1_rewrite_purge_to_get(struct sk_buff *head)
 	return -ENOMEM;
 }
 
+#define CHOP_LEADING_CRLF
+#ifdef CHOP_LEADING_CRLF
+/**
+ * Removes single leading CR or CRLF or LF, if any.
+ * Indicates a buggy message if there are more leading CR or LF or other characters <= 0x20 with EBADMSG.
+ * @return zero on success and error code on error.
+ */
+static int
+tfw_h1_chop_leading_crlf(struct sk_buff **head)
+{
+	struct sk_buff *skb = *head;
+	struct sk_buff *stop = *head;
+	char *p, *q;
+	int i, n, ret, ba;
+	char buf[sizeof(__u32)];
+	struct skb_shared_info *si;
+	skb_frag_t *frag;
+
+	if (likely(skb_headlen(skb) >= 3))
+	{
+		p = q = skb->data;
+		if (unlikely(*p == \r)) p++;
+		if (unlikely(*p == \n)) p++;
+		if (unlikely(*p <= 0x20))
```

Comment:
The parser accepts 10-20-30 CRLFs.

Reply:
I meant that at this place you don't need the `<= 0x20` check because the main parser takes care about this.

> The parser accepts 10-20-30 CRLFs.

Yes, because you removed you previous changes for the main parser. The changes must be restored.